test/functionalTest/cases/1308_subscription_cache/subCacheThrottlingAndExpirationAfterSubUpdateWhenSubExpired.test

```diff
@@ -0,0 +1,330 @@
+# Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Subscription Cache Throttling and Expiration After Sub-Update once the subscription has expired
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 38,101 IPv4 -multiservice -subCacheIval 5
+accumulatorStart
+
+--SHELL--
+
+#
+# 00. Create subscription for s0.* entities, attribute A, w/o tenant, expiration time 1 sec
+# 01. Update the subscription, after it has expired, now setting duration to 10 secs
+# 02. Create entity s01 w/ attr A w/o tenant
+# 03. Modify entity s01 w/ attr A w/o tenant - notification lost by throttling
+# 04. Modify entity s01 w/ attr A w/o tenant - notification lost by throttling
+# 05. Wait 5 secs so that the throttling of 4 secs is OK
+# 06. Modify entity s01 w/ attr A w/o tenant - notification NOT lost by throttling
+# 07. Modify entity s01 w/ attr A w/o tenant - notification lost by throttling
+# 08. Wait 6 secs more, to make subscription expire
+# 09. Modify entity s01 w/ attr A w/o tenant - no notification as expired
+# 10. Dump accumulator - see two notifications
+#
+
+
+echo "00. Create subscription for s0.* entities, attribute A, w/o tenant, expiration time 1 sec"
+echo "========================================================================================="
+payload='{
+  "entities": [
+    {
+      "isPattern": "true",
+      "id": "s0.*"
+    }
+  ],
+  "reference": "http://localhost:'${LISTENER_PORT}'/notify",
+  "duration": "PT1S",
```

Comment:
No problem here either, in step '01. Update the subscription' the sub is put in order